src/resources/views/crud/fields/inc/repeatable_row.blade.php

```diff
@@ -17,31 +17,26 @@
     </div>
     @foreach($field['subfields'] as $subfield)
         @php
-            // make sure the field definition is an array
-            if (is_string($subfield)) {
-                $subfield = ['name' => $subfield];
-            }
-
-            if (!isset($field['model'])) {
-                // we're inside a simple 'repeatable' with no model/relationship, so
-                // we assume all subfields are supposed to be text fields
-                $subfield['type'] = $subfield['type'] ?? 'text';
-                $subfield['entity'] = $subfield['entity'] ?? false;
-            } else {
-                // we should use 'model' as the `baseModel` for all subfields, so that when
-                // we look if `category()` relationship exists on the model, we look on
-                // the model this repeatable represents, not the main CRUD model
-                $subfield['baseModel'] = $subfield['baseModel'] ?? $field['model'];
-            }
-            $subfield = $crud->makeSureFieldHasNecessaryAttributes($subfield);
             $fieldViewNamespace = $subfield['view_namespace'] ?? 'crud::fields';
             $fieldViewPath = $fieldViewNamespace.'.'.$subfield['type'];
 
             if(isset($row)) {
                 if(!is_array($subfield['name'])) {
-                    // this is a fix for 4.1 repeatable names that when the field was multiple, saved the keys with `[]` in the end. Eg: `tags[]` instead of `tags`
-                    if(isset($row[$subfield['name']]) || isset($row[$subfield['name'].'[]'])) {
-                        $subfield['value'] = $row[$subfield['name']] ?? $row[$subfield['name'].'[]'];
+                    if(!Str::contains($subfield['name'], '.')) {
+                        // this is a fix for 4.1 repeatable names that when the field was multiple, saved the keys with `[]` in the end. Eg: `tags[]` instead of `tags`
+                        if(isset($row[$subfield['name']]) || isset($row[$subfield['name'].'[]'])) {
+                            $subfield['value'] = $row[$subfield['name']] ?? $row[$subfield['name'].'[]'];
+                        }
+                        // since repeatable changes names both in JS and PHP we need to make sure we have a name
+                        // that corresponds to the full relation that we can later convert into brakets to use in repeatable
+                        // and also use it as key in repeatable if defined (repeatable over repeatable)
+                        $subfield['real_name'] = $field['name'].'.'.$subfield['name'];
+                        $subfield['name'] = $field['name'].'['.$repeatable_row_key.']['.$subfield['name'].']';
+                        
+                    }else{
+                        $subfield['value'] = \Arr::get($row, $subfield['name']);
+
+                        $subfield['name'] = $field['name'].'['.$repeatable_row_key.']['.Str::replace('.', '][', $subfield['name']).']';
```

Comment:
Is this change needed now, in this PR? Or is it related to https://github.com/Laravel-Backpack/CRUD/pull/4116/files ? Or was this a previous bug, that we didn't know about?